docs/docs/secrets-managers.mdx

```diff
@@ -0,0 +1,142 @@
+---
+id: secrets-managers
+sidebar_label: Secrets Managers
+title: Secrets Managers
+description: Credentials used to authenticate to external services are safely stored in a secrets manager.
+abstract: Credentials managers are used to safeguard credentials used to authenticate to external services.
+---
+
+<!-- this file is version specific, do not use `@site/...` syntax -->
+import RasaProLabel from '@theme/RasaProLabel';
+import RasaProBanner from '@theme/RasaProBanner';
+
+<RasaProLabel />
+<RasaProBanner/>
+
+## Secrets Managers
+Secrets managers are used to safeguard credentials used to authenticate to external services,
+like databases.
+
+Rasa Pro supports the following secrets managers:
+* [HashiCorp Vault](https://www.hashicorp.com/products/vault)
+
+
+Currently, Rasa Pro supports sefeguarding credentials for the following services:
+- [Tracker Stores](./tracker-stores.mdx)
+
+
+## Vault Secrets Manager
+
+The Vault Secrets Manager is used to store credentials used to authenticate to external services.
+The credentials are stored in a Vault instance and can be encrypted at rest.
+
+If you need to encrypt your credentials,
+you can use the [Vault Transit Engine](https://www.vaultproject.io/docs/secrets/transit).
+
+:::note
+Expiring tokens need to be renewed periodically, and renewal process is done over the network,
+we will try to renew the token 5 seconds before it expires.
+If token's TTL is less than 5 seconds, we will try to renew it ASAP,
+but it might fail due to network latency.
+:::
+
+
+### Authentication
+Rasa Pro supports authentication to Vault through
+[Token](https://www.vaultproject.io/docs/auth/token).
+
+Both `expiring` and `non-expiring`(so called, root tokens) tokens are supported.
+Rasa Pro will automatically renew the token if it is expiring.
+
+
+### How to configure access to Vault
+
+Access to Vault secrets manager can be configured with environment variables
+and through `endpoints.yml` configuration file.
+
+Environment variables and `endpoints.yml` configuration file are merged together
+and *the values from the environment variables take precedence*.
+
+Following environment variables are available:
+```bash
+SECRET_MANAGER = "vault" # which secrets manager to use, currently only vault is supported
+VAULT_HOST # address of the vault server
+VAULT_TOKEN # token to authenticate to the vault server
+VAULT_RASA_SECRETS_PATH # path to the secrets in the vault server, if not set it defaults to `rasa-secrets`
+VAULT_TRANSIT_MOUNT_POINT # if transit secrets engine is enabled, set this to  mount point of the transit engine
+```
+
+The `endpoints.yml` configuration for vault secrets manager looks like this:
+```yaml-rasa {3} title="endpoints.yml
+secrets_manager:
```

Comment:
I would add exactly the same comments. If users look at the doc, they might just look at the yml file doc if this is their preferred way, and will have to scroll up to look at the doc for each variable. (I'm talking about comments like "# which secrets manager to use, currently only 'vault' is supported")